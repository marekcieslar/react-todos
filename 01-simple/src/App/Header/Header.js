import React from 'react';

function Header({ title }) {
  return (
    <header>
      <h1 data-testid="header-title">{title}</h1>
    </header>
  );
}

export default Header;
