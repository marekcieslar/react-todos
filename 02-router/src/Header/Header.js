import React, { useContext } from 'react';
import { Link, NavLink } from 'react-router-dom';
import quoteContext from '../contexts/quoteContext';

const activeStyle = {
  color: 'red',
  fontWeight: 'bold',
};

function Header() {
  const [quote] = useContext(quoteContext);

  return (
    <nav>
      <h1>
        <Link to="/">Quotes</Link>
      </h1>
      <span>count: {quote.length}</span>
      <ul>
        <li>
          <NavLink exact to="/" activeStyle={activeStyle}>
            Home
          </NavLink>
        </li>
        <li>
          <NavLink to="/about" activeStyle={activeStyle}>
            About
          </NavLink>
        </li>
        <li>
          <NavLink to="/quote" activeStyle={activeStyle}>
            Quotes
          </NavLink>
        </li>
        <li>
          <NavLink to="/quote-form" activeStyle={activeStyle}>
            Add quote
          </NavLink>
        </li>
      </ul>
    </nav>
  );
}

export default Header;
