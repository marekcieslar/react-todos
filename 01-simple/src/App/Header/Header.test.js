import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

it('should render', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header></Header>, div);
});

it('should render with params', () => {
  const { getByTestId } = render(<Header title="awesome title" />);
  expect(getByTestId('header-title')).toHaveTextContent('awesome title');
});

it('should render with params from variable', () => {
  const title = 'awesome title from variable';
  const { getByTestId } = render(<Header title={title} />);
  expect(getByTestId('header-title')).toHaveTextContent(title);
});

it('matches snapshot', () => {
  const title = 'awesome title for renderer';
  const tree = renderer.create(<Header title={title} />).toJSON();
  expect(tree).toMatchSnapshot();
});
