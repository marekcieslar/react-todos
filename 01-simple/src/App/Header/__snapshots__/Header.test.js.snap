// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header>
  <h1
    data-testid="header-title"
  >
    awesome title for renderer
  </h1>
</header>
`;
